File: colas.py
class Cola:
    def __init__(self, size):
        self.frente = -1
        self.atras = -1
        self.tam = size
        self.arr = [0] * size

    def estavacia(self):
        return self.frente == -1

    def estallena(self):
        return self.atras == self.tam - 1

    def enqueue(self, valor):
        if self.estallena():
            print("Cola llena")
            return
        if self.frente == -1:
            self.frente = 0
        self.atras += 1
        self.arr[self.atras] = valor

    def dequeue(self):
        if self.estavacia():
            print("Cola vacía")
            return -1
        x = self.arr[self.frente]
        if self.frente == self.atras:
            self.frente = self.atras = -1
        else:
            self.frente += 1
        return x

    def peek(self):
        if self.estavacia():
            return -1
        return self.arr[self.frente]


# Ejemplo
q = Cola(5)
q.enqueue(10)
q.enqueue(20)
print("Frente:", q.peek())
print("Dequeue:", q.dequeue())

File: colas.java
public class Cola {
    int frente, atras, tam;
    int[] arr;

    public Cola(int size) {
        frente = -1;
        atras = -1;
        tam = size;
        arr = new int[tam];
    }

    public boolean estaVacia() {
        return frente == -1;
    }

    public boolean estaLlena() {
        return atras == tam - 1;
    }

    public void enqueue(int valor) {
        if (estaLlena()) {
            System.out.println("Cola llena");
            return;
        }
        if (frente == -1) frente = 0;
        arr[++atras] = valor;
    }

    public int dequeue() {
        if (estaVacia()) {
            System.out.println("Cola vacía");
            return -1;
        }
        int x = arr[frente];
        if (frente == atras) frente = atras = -1;
        else frente++;
        return x;
    }

    public int peek() {
        if (estaVacia()) return -1;
        return arr[frente];
    }

    public static void main(String[] args) {
        Cola q = new Cola(5);
        q.enqueue(10);
        q.enqueue(20);
        q.enqueue(30);

        System.out.println("Frente: " + q.peek());
        System.out.println("Dequeue: " + q.dequeue());
    }
}
